pkg/firewall/firewall.go

```diff
@@ -280,6 +298,21 @@ func (soakp *stakeOrActiveKeepPolicy) isKeepActive(
 		return true, nil
 	}
 
+	cache.mutex.Lock()
```

Comment:
Shouldn't we have a separate mutex for each property cached in `keepInfoCache`?

Reply:
I don't think it will help. All those checks need to be executed sequentially so we'll not gain anything from separate mutexes IMO.